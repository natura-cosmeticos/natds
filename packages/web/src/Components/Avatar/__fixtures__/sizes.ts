import { AvatarSize } from '../Avatar.props'

export const sizes : Record<AvatarSize, AvatarSize> = {
  tiny: 'tiny',
  small: 'small',
  standard: 'standard',
  large: 'large',
  huge: 'huge'
}

export default sizes
