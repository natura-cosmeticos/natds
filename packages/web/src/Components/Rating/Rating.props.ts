import * as React from "react";

export type IRatingSizes = "small" | "standard" | "semi" | "semix";

export interface IRatingProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [propName: string]: any;

  /**
   * Forward CSS class name to `class` HTML attribute.
   *
   * @optional
   * @type string
   */
  className?: string;

  /**
   * The input id property
   *
   * @required
   * @type string
   */
  id?: string;

  /**
   * Style to be used on Rating
   */
  style?: React.CSSProperties;

  /**
   * Size of the `Rating`. Defaults to `"small"`.
   *
   * @optional
   */
  size?: IRatingSizes;

  /**
   * The rating value
   *
   * @optional
   */
  rate?: number;
}