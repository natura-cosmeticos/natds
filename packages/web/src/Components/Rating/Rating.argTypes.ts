import { ArgType } from "@storybook/addons";
import { IRatingProps } from "./Rating.props";
import { sizes } from "./__fixtures__/sizes";

export const argTypes : Record<keyof IRatingProps, ArgType> = {
  classes: {
    control: { type: "text" },
    description: "The classes to use on the container 'div'",
    table: {
      type: { summary: "string" },
    },
    type: { name: "string", required: false },
  },
  id: {
    control: { type: "text" },
    description: "The input id property",
    table: {
      type: { summary: "string" },
    },
    type: { name: "string", required: true },
  },
  style: {
    control: { type: "object" },
    description: "Inline style for the container 'div'",
    table: {
      type: { summary: "React.CSSProperties" },
    },
    type: { name: "object", required: false },
  },
  size: {
    control: {
      options: sizes,
      type: "select",
    },
    description: "The size of the Rating",
    table: {
      defaultValue: { summary: "\"small\"" },
      type: { summary: "'small' \\| 'standard' \\| 'semi' \\| 'semix'" },
    },
  },
  rate: {
    control: {
      type: "number",
    },
    description: "The rating value",
    table: {
      type: { summary: "Value between 0 and 5" },
    },
  },
};

export default argTypes;