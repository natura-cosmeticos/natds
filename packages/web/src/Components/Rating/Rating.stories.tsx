import { Meta, Story } from "@storybook/react";
// eslint-disable-next-line no-use-before-define
import * as React from "react";
import { Rating } from "./Rating";
import { IRatingProps } from "./Rating.props";
import { argTypes } from "./Rating.argTypes";
import { sizes } from "./__fixtures__/sizes";

export default {
  argTypes,
  component: Rating,
  title: "Components/Rating",
} as Meta;

const Template : Story<IRatingProps> = (args: IRatingProps) => <Rating />;

export const Playground : Story<IRatingProps> = Template.bind({});
Playground.args = {
  id: "playground",
  size: sizes.small,
  rate: 3
};