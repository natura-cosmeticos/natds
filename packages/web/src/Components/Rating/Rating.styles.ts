import makeStyles from "@material-ui/core/styles/makeStyles";
import { createStyles } from "@material-ui/core";
import { IThemeWeb } from "../../Themes";
import { IRatingProps, IRatingSizes } from "./Rating.props";

type RatingStylesArgs = {
  size?: IRatingSizes,
  theme: IThemeWeb
};

export const getFontSize : ({ size, theme }: RatingStylesArgs) => string = ({ size = "small", theme }) => {
  const { sizes } = theme;
  const nullValue = "";

  if (size && sizes && sizes[size]) {
    const fontSize = sizes[size];

    return `${fontSize}px`;
  }

  return nullValue;
};

const style = (theme: IThemeWeb) => createStyles({
  root: ({ size }: IRatingProps) => ({
    fontSize: getFontSize({ size, theme }),
  }),

  iconEmpty:{
    color: theme.palette?.text?.secondary,
  }
});

export const useStyles = makeStyles(style, { name: "NatDSRating" });

export default useStyles;