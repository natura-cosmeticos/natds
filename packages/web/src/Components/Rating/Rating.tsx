// eslint-disable-next-line no-use-before-define
import * as React from "react";
import MaterialRating from "@material-ui/lab/Rating";
import clsx from "clsx";
import { IRatingProps } from "./Rating.props";
import useStyles from "./Rating.styles";

export { IRatingProps } from "./Rating.props";

/**
 * ## Importing
 *
 * ```
 * import { Rating } from '@naturacosmeticos/natds-web';
 * ```
 */

export const Rating = React.forwardRef<HTMLDivElement, IRatingProps>(
  (props: IRatingProps, ref) => {
    const {
      className,
      size,
      rate
    } = props;

    const classes = useStyles({ size });

    const classNames = clsx([
      className,
      classes.root,
    ]);

    return <MaterialRating
      className={classNames}
      value={rate}
      name="rating"
      readOnly={true}
      ref={ref}
    />;
  });

Rating.displayName = "Rating";

export default Rating;