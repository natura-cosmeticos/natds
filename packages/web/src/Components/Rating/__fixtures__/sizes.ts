import { IRatingSizes } from "../Rating.props";

export const sizes : Record<IRatingSizes, IRatingSizes> = {
  small: "small",
  standard: "standard",
  semi: "semi",
  semix: "semix"
};

export default sizes;