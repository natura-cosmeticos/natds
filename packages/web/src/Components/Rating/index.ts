import Rating from "./Rating";

export { IRatingProps } from "./Rating.props";
export { Rating };
export default Rating;