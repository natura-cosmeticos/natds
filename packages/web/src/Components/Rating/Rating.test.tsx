import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { sizes } from "./__fixtures__/sizes";
import { Rating } from "./Rating";
import { Provider } from "../../Provider";
import { themes } from "../../Themes";
import { IRatingSizes } from "./Rating.props";

describe("Rating component", () => {
  describe("rendering", () => {

    let testRenderer : TestRenderer.ReactTestRenderer = {} as TestRenderer.ReactTestRenderer;

    it("it should render each size", () => {
      Object.values(sizes).forEach((size: IRatingSizes) => {
        testRenderer = TestRenderer.create(<Rating id={size} size={size}>Mock</Rating>);

        expect(testRenderer).toMatchSnapshot("Rating Sizes");
      });
    });
    it("it should render according to theme", () => {
      testRenderer = TestRenderer.create(<Provider theme={themes.natura.dark}><Rating id={"with-theme"}>Mock</Rating></Provider>);

      expect(testRenderer).toMatchSnapshot("Rating Custom Theme");
    });
  });
});