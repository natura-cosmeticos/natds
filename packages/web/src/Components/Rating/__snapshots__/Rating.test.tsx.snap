// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component rendering it should render according to theme: Rating Custom Theme 1`] = `
<span
  aria-label="null Stars"
  className="MuiRating-root NatDSRating-root-1 NatDSRating-root-3 MuiRating-readOnly"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
</span>
`;

exports[`Rating component rendering it should render each size: Rating Sizes 1`] = `
<span
  aria-label="null Stars"
  className="MuiRating-root NatDSRating-root-1 NatDSRating-root-3 MuiRating-readOnly"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
</span>
`;

exports[`Rating component rendering it should render each size: Rating Sizes 2`] = `
<span
  aria-label="null Stars"
  className="MuiRating-root NatDSRating-root-1 NatDSRating-root-4 MuiRating-readOnly"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
</span>
`;

exports[`Rating component rendering it should render each size: Rating Sizes 3`] = `
<span
  aria-label="null Stars"
  className="MuiRating-root NatDSRating-root-1 NatDSRating-root-5 MuiRating-readOnly"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
</span>
`;

exports[`Rating component rendering it should render each size: Rating Sizes 4`] = `
<span
  aria-label="null Stars"
  className="MuiRating-root NatDSRating-root-1 NatDSRating-root-6 MuiRating-readOnly"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty NatDSRating-iconEmpty-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
</span>
`;
